lesson19/js/components/slider.js

```diff
@@ -0,0 +1,159 @@
+  imagesData.forEach((imageData, index) => {
+    const ListImageItemImg = createAttributedElements("img", {
+      src: imageData.img,
+      alt: imageData.description,
+    });
+    const ListImageItem = createAttributedElements("li", {
+      class: "slide-show_list",
+    });
+    imagesFragment.appendChild(ListImageItem).appendChild(ListImageItemImg);
+    index === slider.state.currentIndex &&
+      ListImageItem.classList.add("is-active");
+  });
+  return imagesFragment;
+};
+
+const getSliderStatusCurrentPagination = () => {
+  slider.state.imageLists = [...document.querySelectorAll(".slide-show_list")];
+  slider.state.currentPagination = createAttributedElements(
+    "p",
+    {
+      class: "pagination",
+    },
+    `${slider.state.currentIndex + 1}/${slider.state.imageLists.length}`
+  );
+  return slider.state.currentPagination;
+};
+
+const getPaginationButtonContainer = () => {
+  const container = createAttributedElements("div", {
+    class: "button-pagination_container",
+  });
+  const imageListsLength = slider.state.imageLists.length;
+  for (let i = 0; i < imageListsLength; i++) {
+    const button = createAttributedElements("button", {
+      class: "pagination-button button-pagination",
+      "data-index": i,
+    });
+    if (i === 0) button.classList.add("is-active");
+    button.addEventListener("click", clickPaginationButtons);
+    container.appendChild(button);
+  }
+  return container;
+};
+
+const getdirectionButtonsFragment = () => {
+  const directions = Object.values(slider.directions);
+  const directionButtonsFragment = document.createDocumentFragment();
+  directions.forEach((direction) => {
+    const button = createAttributedElements("button", {
+      class: `button __${direction}`,
+      id: `js-slider-${direction}`,
+      "data-direction": direction,
+    });
+    button.disabled = direction === "previous";
+    button.addEventListener("click", clickDirectionButtons);
+    directionButtonsFragment.appendChild(button);
+  });
+  return directionButtonsFragment;
+};
+
+const clickPaginationButtons = (e) => {
+  const activeElements = [...document.querySelectorAll(".is-active")];
+  activeElements.forEach((activeElement) =>
+    activeElement.classList.remove("is-active")
+  );
+  slider.state.imageLists[e.target.dataset.index].classList.add("is-active");
+  e.target.classList.add("is-active");
+  slider.state.currentIndex = Number(e.target.dataset.index);
+  controlSlidersMovement();
+  autoPlay.reset();
+};
+
+const clickDirectionButtons = (e) => {
+  const activeElement = document.querySelector(".is-active");
+  activeElement.classList.remove("is-active");
+  const direction = `${e.target.dataset.direction}ElementSibling`;
+  e.target.dataset.direction === slider.directions.Next
+    ? ++slider.state.currentIndex
+    : --slider.state.currentIndex;
+  activeElement[direction].classList.add("is-active");
+  controlSlidersMovement();
+  autoPlay.reset();
+};
+
+const paginationUpdate = () => {
```

Comment:
Q.
Personally, I found it easier to understand function names with the verb first.
```suggestion
const updatePagination = () => {
```

also  directionButtonsControl() and paginationButtonsControl() :)
How about it?

Reply:
You are right! Thanks for letting me know! Fixed.
[9d46c58](https://github.com/443n0511/js-lesson/pull/41/commits/9d46c580061b734c72fea6cae22fef002f108253)

Reply:
Thank you for your fixed!
I figured 'directionButtonsControl()' and  'paginationButtonsControl()' would be the same.

Reply:
I'm sorry. I missed it. These have been revised. Can you please check again?
[c1a7efa](https://github.com/443n0511/js-lesson/pull/41/commits/c1a7efa857516edda26816b9262544c452e0fd41)